src/Analysis/Ast/Impl/Modules/CompiledPythonModule.cs

```diff
@@ -73,34 +75,29 @@ public override string Documentation
                 return string.Empty;
             }
 
-            var sb = new StringBuilder();
-            using (var proc = new ProcessHelper(
-                Interpreter.Configuration.InterpreterPath,
-                args,
-                Interpreter.Configuration.LibraryPath
-            )) {
-                proc.StartInfo.StandardOutputEncoding = Encoding.UTF8;
-                proc.OnOutputLine = s => sb.AppendLine(s);
-                proc.OnErrorLine = s => Log?.Log(TraceEventType.Warning, "Scrape", s);
-
-                Log?.Log(TraceEventType.Verbose, "Scrape", proc.FileName, proc.Arguments);
-
-                proc.Start();
-                var exitCode = proc.Wait(60000);
-
-                if (exitCode == null) {
-                    proc.Kill();
-                    Log?.Log(TraceEventType.Warning, "ScrapeTimeout", proc.FileName, proc.Arguments);
-                    return string.Empty;
-                }
-
-                if (exitCode != 0) {
-                    Log?.Log(TraceEventType.Warning, "Scrape", "ExitCode", exitCode);
-                    return string.Empty;
-                }
-            }
-
-            return sb.ToString();
+            var startInfo = new ProcessStartInfo { 
+                FileName = Interpreter.Configuration.InterpreterPath,
+                Arguments = args.AsQuotedArguments(),
+                WorkingDirectory = Interpreter.Configuration.LibraryPath,
+                UseShellExecute = false,
+                ErrorDialog = false,
+                CreateNoWindow = true,
+                StandardOutputEncoding = Encoding.UTF8,
+                RedirectStandardInput = true,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true
+            };
+            var ps = Services.GetService<IProcessServices>();
+
+            Log?.Log(TraceEventType.Verbose, "Scrape", startInfo.FileName, startInfo.Arguments);
+            Task<IReadOnlyList<string>> task = null;
+
+            try {
+                task = ps.ExecuteAndCaptureOutputAsync(startInfo);
```

Comment:
IMO, it is better to pass timeout as parameter to `ExecuteAndCaptureOutputAsync`:
```cs
try {
    var token = new CancellationTokenSource(60000).Token;
    var lines = ps.ExecuteAndCaptureOutputAsync(startInfo, token).GetAwaiter().GetResult();
    return string.Join(Environment.NewLine, lines);
} catch (Exception) {
    return string.Empty;
}
```

Reply:
I can create cancellation token with timeout instead

Reply:
Done